DigitalLibrary/DigitalLibrary/MainScreens/LoginScreen/LoginViewModel.swift

```diff
@@ -0,0 +1,69 @@
+//
+//  LoginViewModel.swift
+//  DigitalLibrary
+//
+//  Created by Desislav Adamov on 4.02.24.
+//
+
+import SwiftUI
+
+final class LoginViewModel: ObservableObject {
+    @ObservedObject private var appRootManager: AppRootManager
+
+    private(set) var authenticationProvider: AuthenticationProvidable
+    private(set) var userProvider: UserProvidable
+
+    private var startedEditingEmail = false
+    private var startedEditingPassword = false
+
+    @Published var emailIsValid = true {
+        didSet {
+            if emailIsValid == true {
+                startedEditingEmail = true
+            }
+        }
+    }
+
+    @Published var passwordIsValid = true {
+        didSet {
+            if passwordIsValid == true {
+                startedEditingPassword = true
+            }
+        }
+    }
+
+    @Published var email: String = ""
+    @Published var password: String = ""
+    
+    var formIsValid: Bool {
+        emailIsValid && passwordIsValid && startedEditingEmail && startedEditingPassword
+    }
+
+    init(appRootManager: AppRootManager,
+         authenticationProvider: AuthenticationProvidable,
+         userProvider: UserProvidable) {
+        self.appRootManager = appRootManager
+        self.authenticationProvider = authenticationProvider
+        self.userProvider = userProvider
+    }
+
+    func validateEmail() -> String {
+        email != "" ? "" : "Required"
+    }
+
+    func validatePassword() -> String {
+        password != "" ? "" : "Required"
+    }
+
+    func signIn() {
+        Task {
+            do {
+                try await authenticationProvider.signIn(emailAddress: email, password: password)
+
+                appRootManager.currentRoot = .main
+            } catch {
+                print(error)
```

Comment:
I think we can handle these errors that occur with a popup, and since we already have this functionality from the iOS standard library, I think it will be pretty easy to use it.

Reply:
Do you mean to create a popup with value OK and title the error title?

Reply:
Yeah, I mean we do not want to have print statements and it is good practise to have error handling. And to be more user friendly I think the user will want to know when the error happens and I think something like popup with error title and buttons like Dismiss for example or Try again and Cancel. You may decide which is more appropriate.

Reply:
I think just OK is the most appropriate. Try again is not appropriate for some scenarios. E.g. if you enter invalid credentials and you try again, you would always get the same error. Cancel is also not what the 
user actually does
<img width="325" alt="Screenshot 2024-02-12 at 9 38 26" src="https://github.com/SwiftFMI/2024_prj_DigitalLibrary/assets/89339207/3042b493-f814-4ce2-bebd-24cae140bada">